proposals/slottable-request.md

```diff
@@ -0,0 +1,86 @@
+# Slottable Request Protocol
+aka "Render Props for Web Components"
+
+**Author**: Kevin Schaaf (@kevinpschaaf, Google)
+
+**Status**: Draft
+
+**Last Updated**: 2023-08-28
+
+
+# Background
+
+There are often use cases that require a component to render UI based on data, but where the specific rendering of the data should be controllable by the user. Use cases of this include:
+
+*   A data table or tree component, which manages the rendering of cells or tree nodes, but where the given look and feel of the items is customizable by the user.
+*   A virtual list component, which optimizes rendering a list of data by only rendering a subset of user-templated items based on which array instances are visible on the screen.
+*   Components that fetch data themselves, but provide the ability for the user to customize the rendering of the data.
```

Comment:
One use case I'd like to see mentioned is for deferred rendering. ie, a tooltip or menu component might to render some user-provided DOM, but only rarely and on-demand. Using this protocol to trigger that render would reduce the number of elements unnecessarily rendered.

Reply:
Agreed, this should be  noted as part of the fundamental motivation. In particular, I believe this protocol addresses *most* of the [`slot` related criticisms levied by Vue](https://vuejs.org/guide/extras/web-components.html#web-components-vs-vue-components):
1. Slottable content can be rendered _lazily_, based on receiving the `slottable-request` event. It no longer needs to be rendered _eagerly_.
2. Data can be passed to the slottable content generator to be used during rendering.

Reply:
Added.